lib/uring/uring_stubs.c

```diff
@@ -146,6 +148,34 @@ ocaml_uring_sq_ready(value v_uring) {
   return (Int_val(io_uring_sq_ready(ring)));
 }
 
+void /* noalloc */
+ocaml_uring_set_timespec(value v_sketch_ptr, value v_timeout)
+{
+  struct __kernel_timespec *t = Sketch_ptr_val(v_sketch_ptr);
+  t->tv_sec = 0;
+  t->tv_nsec = Int64_val(v_timeout);
+}
+
+value /* noalloc */
+ocaml_uring_submit_timeout(value v_uring, value v_id, value v_sketch_ptr, value v_clock)
+{
+  struct __kernel_timespec *t = Sketch_ptr_val(v_sketch_ptr);
+  struct io_uring* ring = Ring_val(v_uring);
+  struct io_uring_sqe* sqe;
+  int clock;
+
+  if (v_clock == caml_hash_variant("Boottime"))
+    clock = IORING_TIMEOUT_BOOTTIME;
+  else
+    clock = IORING_TIMEOUT_REALTIME;
+
+  sqe = io_uring_get_sqe(ring);
+  if (!sqe) return Val_false;
+  io_uring_prep_timeout(sqe, t, 0, clock);
```

Comment:
> Uring.timeout job returns -62. Is this optimal/ergonomic?

```suggestion
  io_uring_prep_timeout(sqe, t, 0, clock | IORING_TIMEOUT_ETIME_SUCCESS);
```
It would be good to support `IORING_TIMEOUT_ABS` too (as a flag).

Reply:
Done.

Reply:
using `IORING_TIMEOUT_ETIME_SUCCESS` seems to give EINVAL (-22). https://ci.ocamllabs.io/github/ocaml-multicore/ocaml-uring/commit/41f87340200483507ce5050f1a6ba86e67ab47c3/variant/debian-11-4.12_opam-2.1

Perhaps the ocaml-ci is not using the supported version of uring. In my local test, even after using `IORING_TIMEOUT_TIME_SUCCESS` it still returns -65.

Reply:
It looks like `IORING_TIMEOUT_ETIME_SUCCESS` doesn't change the result value, it just doesn't mark the job as failed for chaining. And also requires Linux 5.16. So we should revert this change and let the caller handle it.